### PR TITLE
Fixes #42407 add inbound_nat_rules for loadbalancer

### LINKED ISSUE #42407: Missing parameter in azure_rm_loadbalancer
##### SUMMARY

In inbound_nat_pools is not defined service type parameter in azure side we can choose service type.  
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME

azure_rm_loadbalancer
##### ANSIBLE VERSION

```
2.5
```

##### STEPS TO REPRODUCE

```yaml

```

##### ACTUAL RESULTS

```

```

### PR DESCRIPTION
**Pull request by yuwzho**
##### SUMMARY

Fixes #42407 add inbound_nat_rules for loadbalancer

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

azure_rm_loadbalancer

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py [hunk 1]
```diff
@@ -204,6 +204,54 @@
...
+            name:
+                description: name of the inbound nat rule.
+                required: True
+            frontend_ip_configuration:
+                description: A reference to frontend IP addresses.
+                required: True
+            protocol:
+                description: IP protocol for the inbound nat rule.
+                choices:
+                    - Tcp
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py:221]: lower case?

File: lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py [hunk 2]
```diff
@@ -746,14 +862,15 @@ def exec_module(self, **kwargs):
...
             )
 
             if load_balancer:
+                self.new_load_balancer = self.object_assign(self.new_load_balancer, load_balancer)
+                load_balancer_dict = load_balancer.as_dict()
                 new_dict = self.new_load_balancer.as_dict()
-                if (self.location != load_balancer['location'] or
-                        self.sku != load_balancer['sku']['name'] or
-                        not default_compare(new_dict, load_balancer, '')):
+                if not default_compare(new_dict, load_balancer_dict, ''):
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py:867]: use set protocol to udp at first run, then not set the value at second run, default will be tcp, compare fail, actually user don't want to change the value

Comment (yuwzho) [reply lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py]: Here I keep consistent with all other protocol in this module. I can remove all the default value for it anyway.

**Review by yungezz — APPROVED** [review]

**Review by yungezz — COMMENTED** [review]
there're still comments

**Review by yungezz — CHANGES_REQUESTED** [review]
comments mainly on default value not updated

**Review by yungezz — APPROVED** [review]

File: lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py (hunk 2 above)
Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py:867]: if parameter is None in the second run, it won't be included in comparison. so this scenario is ok.

File: lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py [hunk 3]
```diff
@@ -807,10 +918,32 @@ def create_or_update_load_balancer(self, param):
         try:
             poller = self.network_client.load_balancers.create_or_update(self.resource_group, self.name, param)
             new_lb = self.get_poller_result(poller)
-            return new_lb.as_dict()
+            return new_lb
         except CloudError as exc:
             self.fail("Error creating or updating load balancer {0} - {1}".format(self.name, str(exc)))
 
+    def object_assign(self, patch, origin):
+        attribute_map = set(self.network_models.LoadBalancer._attribute_map.keys()) - set(self.network_models.LoadBalancer._validation.keys())
```
Comment (zikalino) [inline lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py:926]: we really need something generic.... getting complex :-)

**Review by zikalino — APPROVED** [review]

**Comment by zikalino** [comment]
I think the change looks good.
although this module needs some general refactoring, but can't be done in this change.

**Pull request merged**